include/odp/api/spec/pool.h

```diff
@@ -44,6 +44,57 @@ extern "C" {
 /** Maximum number of packet pool subparameters */
 #define ODP_POOL_MAX_SUBPARAMS  7
 
+/**
+ * Pool statistics counter options bit field
+ *
+ * Pool statistics counter options listed in a bit field structure.
+ * This structure shall be used to get the supported counters for the pool
+ * and configure the statistics counters.
+ *
+ * @see odp_pool_capability_t::buf::stats_supported
+ * @see odp_pool_param_t::stats_config
+ *
+ */
+typedef union odp_pool_stats_opt_t {
+	/** Option flags */
+	struct {
+		/** Select the odp_pool_stats_t::available counter */
+		uint64_t available             : 1;
+
+		/** Select the odp_pool_stats_t::available_local_cache counter */
+		uint64_t available_local_cache : 1;
+
+		/** Select the odp_pool_stats_t::ops counter */
+		uint64_t ops                   : 1;
+	} bit;
+
+	/** All bits of the bit field structure
+	  *
+	  * This field can be used to set/clear all flags, or bitwise
+	  * operations over the entire structure. */
+	uint64_t all;
+} odp_pool_stats_opt_t;
+
+/**
+ * Pool statistics
+ */
+typedef struct odp_pool_stats_t {
+	/**
+	 * The number of available entries in the pool.
+	 */
+	uint64_t available;
+
+	/**
+	 * The number of available entries in the local cache of all the CPU of a pool.
+	 */
+	uint64_t available_local_cache;
+
+	/**
+	 * The number of free and alloc operations in the pool.
+	 */
+	uint64_t ops;
```

Comment:
This seems a bit vague. Should we instead have separate counters for allocs and frees?

Reply:
On OcteonTx HW, we have only a single counter for both.(It is the total number of free and alloc operation happened during the life span of a pool). Any thoughts?

Reply:
OK. Thinking more about it, it could be useful to have separate counters for the global pool and local caches. 
E.g.
```
/**
 * Pool statistics
 */
typedef struct odp_pool_stats_t {
	/** The number of available entries in the pool */
	uint64_t available;

	/** The number of alloc operations from the pool */
	uint64_t alloc_ops;

	/** The number of free operations to the pool */
	uint64_t free_ops;

	/** The combined number of alloc and free operations */
	uint64_t combined_ops;

	/** Local cache statistics */
	struct {
		/** The number of available entries in the local caches of all threads using the
		 *  pool */
		uint64_t available;

		/** The number of alloc operations from the pool caches  */
		uint64_t alloc_ops;

		/** The number of free operations to the pool caches */
		uint64_t free_ops;

		/** The combined number of alloc and free operations */
		uint64_t combined_ops;

	} cache;

} odp_pool_stats_t;
```

Reply:
Looks good. I think, it is better to have cache_* variable instead of the new struct as opt_s is bit field.

Reply:
True, this is the version I'm currently using for the implementation:
```
/**
 * Pool statistics
 */
typedef struct odp_pool_stats_t {
	/** The number of available entries in the pool */
	uint64_t available;

	/** The number of alloc operations from the pool */
	uint64_t alloc_ops;

	/** The number of free operations to the pool */
	uint64_t free_ops;

	/** The combined number of alloc and free operations */
	uint64_t combined_ops;

	/** The number of available entries in the local caches of all threads
	 *  using the pool */
	uint64_t cache_available;

	/** The number of alloc operations from the pool caches  */
	uint64_t cache_alloc_ops;

	/** The number of free operations to the pool caches */
	uint64_t cache_free_ops;

	/** The combined number of alloc and free operations */
	uint64_t cache_combined_ops;

} odp_pool_stats_t;
```

Reply:
Looks good. The last comment is missing "from the pool caches"